python/cudf/cudf/core/buffer.py

```diff
@@ -110,7 +96,20 @@ def deserialize(cls, header, frames):
         obj : Buffer
             Returns an instance of Buffer.
         """
-        return Buffer(frames[0], header=header)
+        assert len(frames) == 1, "Use Buffer.serialize() for serialization"
+        iface = header["cuda_array_interface"]
+        if iface["shape"][0] > 0:
+            arr = numba.cuda.devicearray.DeviceNDArray(
```

Comment:
Should this be an `rmm` backed device array instead?

Reply:
@shwina probably.  Is that `rmm.to_device` ?

Reply:
Good point, fixed

Reply:
Looks like it's still a numba devicearray?